python/taichi/lang/ext_array.py

```diff
@@ -38,3 +38,59 @@ def loop_range(self):
             taichi_core.Expr: See above.
         """
         return self.ptr
+
+
+class AnyArray:
+    """Class for arbitrary arrays in Python AST.
+
+    Args:
+        ptr (taichi_core.Expr): A taichi_core.Expr wrapping a taichi_core.ExternalTensorExpression.
+        element_shape (Tuple[Int]): () if scalar elements (default), (n) if vector elements, and (n, m) if matrix elements.
+        is_soa (bool): Whether to change the layout from AOS (default) to SOA.
+    """
+    def __init__(self, ptr, element_shape, is_soa):
```

Comment:
I'd suggest to use `layout`, so that we can have  a wider range of options (e.g. AOS, SOA, AoSoA)

Reply:
Well the reason why I use bool instead of ti.AOS/SOA is circular import...

Reply:
Can we move around `ti.AOS/SOA` to break the cycle?

Reply:
I cannot use `ti.AOS/SOA` as default values of method parameters. It seems that the Matrix class hacks it using `None` as default value and changes it into `ti.AOS` in the method...

Reply:
If we want use `ti.xxx` then no. If we want to move it to another file and directly use `taichi.lang.xxx.xx` then possible.

Reply:
> directly use taichi.lang.xxx.xx then possible.

SG!

Reply:
Shall we merge this first and add a separate PR to do this? This is a slight refactoring which will affect a couple of places. I plan to make an Enum for Layout instead of a class with a true/false var inside (current implementation). This will allow us to support more potential layouts in the future. Furthermore, should we expose this to users as `ti.Layout.AOS` or keep the current `ti.AOS`?